src/rezplugins/build_system/python.py

```diff
@@ -0,0 +1,325 @@
+"""
+Built-in simple python build system
+"""
+import os
+import glob
+import shutil
+import subprocess
+import sys
+import functools
+import argparse
+from pipes import quote
+
+from rez.build_system import BuildSystem
+from rez.build_process_ import BuildType
+from rez.util import create_forwarding_script
+from rez.packages_ import get_developer_package
+from rez.resolved_context import ResolvedContext
+from rez.config import config
+from rez.utils.colorize import heading, Printer
+from rez.utils.logging_ import print_warning
+from rez.utils.yaml import dump_yaml
+
+
+class PythonBuildSystem(BuildSystem):
+    """The standard python setup.py build system.
+
+    This is a wrapper around the normal python setup.py build system.  It
+    allows python projects to seamlessly build using rez without having to
+    create a cmake or bez build wrapper.
+
+    """
+    @classmethod
+    def name(cls):
+        return "python"
+
+    @classmethod
+    def is_valid_root(cls, path):
+        return (os.path.isfile(os.path.join(path, "setup.py"))
+                and not os.path.isfile(os.path.join(path, "rezbuild.py")))
+
+    def __init__(self, working_dir, opts=None, package=None, write_build_scripts=False,
+                 verbose=False, build_args=[], child_build_args=[]):
```

Comment:
Is the use of mutable defaults here intentional? I'm referring to `build_args=[]` and `child_build_args=[]`. From my understanding, this is an anti-pattern that can lead to hard-to-find-bugs:

- https://docs.python-guide.org/writing/gotchas/
- https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html

Could we achieve the same thing using something like:

```python
 def __init__(self, working_dir, opts=None, package=None, write_build_scripts=False,
                 verbose=False, build_args=None, child_build_args=None):
        super(PythonBuildSystem, self).__init__(
            working_dir,
            opts=opts,
            package=package,
            write_build_scripts=write_build_scripts,
            verbose=verbose,
            build_args=build_args or [],
            child_build_args=child_build_args or [])
```